src/data_hub/dataframe.py

```diff
@@ -81,66 +85,92 @@ def __setitem__(self, key, value):
+            date_str = self.attrs["last_updated_row"].date_updated
+            date_obj = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f").replace(
+                tzinfo=timezone.utc
+            )
+            edited_time_ago = humanize.naturaltime(now - date_obj)
 
-        header = f'<h4 style="color: #363636;">Deep Origin / {org_name} / <a href = "{url}">{name} </a></h4>'
-        txt = f'<p style="font-size: 12px; color: #808080;">Created {created_time_ago}. Row {self.attrs["last_updated_row"].hid} was last edited {edited_time_ago}'
-        try:
-            last_edited_by = get_last_edited_user_name(self.attrs["last_updated_row"])
-            txt += "  by " + last_edited_by + ".</p>"
+            header = f'<h4 style="color: #808080;">Deep Origin / {org_name} / <a href = "{url}">{name} </a></h4>'
+            txt = f'<p style="font-size: 12px; color: #808080;">Created {created_time_ago}. Row {self.attrs["last_updated_row"].hid} was last edited {edited_time_ago}'
+            try:
+                last_edited_by = get_last_edited_user_name(
+                    self.attrs["last_updated_row"]
+                )
+                txt += "  by " + last_edited_by + ".</p>"
+            except Exception:
+                # give up. this should not cause the dataframe to
+                # not print.
+                txt += ".</p>"
+
+            if self._modified_columns:
+                txt += '<p style="color: #808080; font-size: 12px">⚠️ This dataframe contains changes that have not been written back to the Deep Origin database.</p>'
+            elif self.auto_sync:
+                txt += '<p style="color: #808080; font-size: 12px">🧬 This dataframe will automatically write changes made to it back to Deep Origin.</p>'
+            df_html = super()._repr_html_()
+            return header + txt + df_html
         except Exception:
-            # give up. this should not cause the dataframe to
-            # not print.
-            txt += ".</p>"
-
-        if self._modified_columns:
-            txt += '<p style="color: #808080; font-size: 12px">⚠️ This dataframe contains changes that have not been written back to the Deep Origin database.</p>'
-        elif self.auto_sync:
-            txt += '<p style="color: #808080; font-size: 12px">🧬 This dataframe will automatically write changes made to it back to Deep Origin.</p>'
-        df_html = super()._repr_html_()
-        return header + txt + df_html
+            return super()._repr_html_()
 
     def __repr__(self):
         """method override to customize printing in an interactive session"""
 
-        header = f'{self.attrs["metadata"]["hid"]}\n'
-        df_representation = super().__repr__()
-        return header + df_representation
+        try:
+            header = f'{self.attrs["metadata"]["hid"]}\n'
+            df_representation = super().__repr__()
+            return header + df_representation
+        except Exception:
+            return super().__repr__()
```

Comment:
Why would this throw?

Reply:
Could this be slightly refactored to minimize the scope of the try? 

```python
df_representation = super().__repr__()
try:
    header = f'{self.attrs["metadata"]["hid"]}\n'
    return header + df_representation
except Exception:
    return df_representation
```

Reply:
> Why would this throw?

if somehow `attrs` didn't contain the right fields

Reply:
fixed